web-common/src/features/entity-management/watch-resources-client.ts

```diff
@@ -0,0 +1,92 @@
+import { WatchRequestClient } from "@rilldata/web-common/runtime-client/watch-request-client";
+import {
+  getRuntimeServiceGetResourceQueryKey,
+  getRuntimeServiceListResourcesQueryKey,
+  V1Resource,
+  V1WatchResourcesResponse,
+} from "@rilldata/web-common/runtime-client";
+import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import type { QueryClient } from "@tanstack/svelte-query";
+import { get } from "svelte/store";
+
+export function watchResourcesClient(queryClient: QueryClient) {
+  return new WatchRequestClient<V1WatchResourcesResponse>(
+    (runtime) =>
+      `${runtime.host}/v1/instances/${runtime.instanceId}/resources/-/watch`,
+    (res) => invalidateResourceResponse(queryClient, res),
+    () => invalidateAllResources(queryClient)
+  ).start();
+}
+
+function invalidateResourceResponse(
```

Comment:
This is more `handleResourceResponse`, the invalidation comes later in your utility functions below